cmsController.php

```diff
@@ -0,0 +1,195 @@
+    $query->execute();
+    return $query->fetchall();
+}
+
+/**
+ * Takes an array of $items and turns into a string to make a drop-down list
+ *
+ * @param $items array of items (options) to be made into a drop-down
+ * @return string of options
+ */
+function makeDropDown (array $items): string {
+    $dropDownString = "";
+    foreach ($items as $item) {
+        $dropDownString .= '<option value="' . $item['title'] . '">' . $item['title'] . '</option>';
+    }
+    return $dropDownString;
+}
+
+/**
+ * Takes selected item POSTdata and selects values from DB based on POST value
+ *
+ * @param $db PDO to select from
+ * @param $postData assoc array  user response for item to select
+ * @return array of item values
+ */
+function portFolioFill (PDO $db, array $postData):array {
+    $selectedItem=$postData['itemSelect'];
+    $query=$db->prepare("SELECT `title`,`description`,`imgRef`,`projURL`,`github`,`images`.`URL`
+                               FROM `portfolioItems`
+                               LEFT JOIN `images`
+                               ON `portfolioItems`.`imgRef`
+                               =`images`.`id`
+                               WHERE `title`='$selectedItem' && `title` != 'submitPF';");
+    $query->execute();
+    return $query->fetchall();
+}
+
+/**
+ * Gets list of articles from DB
+ *
+ * @param $db PDO to select from
+ * @return array of articles
+ */
+function ArticleList (PDO $db):array {
+    $query=$db->prepare("SELECT `title` FROM `articles` WHERE `deleted`!=1 ");
+    $query->execute();
+    return $query->fetchall();
+}
+
+/**
+ * Takes selected article POSTdata and selects values from DB based on POST value
+ *
+ * @param $db PDO to select from
+ * @param $postData assoc array userresponse for article to select
+ * @return array
+ */
+function SelectArt (PDO $db, array $postData):array {
+    $artSelect = $postData['artSelect'];
+    $query = $db->prepare("SELECT `id`,`title`,`description`, `URL`
+                               FROM `articles`
+                               WHERE `title`='$artSelect';");
+    $query->execute();
+    return $query->fetchall();
+}
+
+/**
+ * Updates About sections on DB
+ *
+ * @param $postData Array used to test if function should run and informs what content should be added to DB
+ * @param $db to add to
+ */
+function updateAbout (array $postData,PDO $db) {
+    if(array_key_exists('submitAbout',$postData)) {
+        foreach($postData as $keys=>$values) {
+            $query = $db->prepare("REPLACE INTO `staticContent`(`name`,`content`) VALUES (?,?)");
+            $query->execute([$keys, $values]);
+        }
+    }
+}
+
+/**
+ * Updates/adds Portfolio item based on Name
+ *
+ * @param $postData array used to test if function should run and informs what content should be added to DB
+ * @param $db to add to
+ */
+function updatePortfolio (array $postData,PDO $db) {
+    if(array_key_exists('submitPf',$postData)) {
```

Comment:
no validation on if the array keys exist, pfTitle, pfDesc, pfURL etc

Reply:
Resolved.